core/base/explicitTriangulation/ExplicitTriangulation.cpp

```diff
@@ -123,21 +123,29 @@ int ExplicitTriangulation::preconditionBoundaryEdgesInternal() {
 #ifdef TTK_ENABLE_MPI
 
 int ExplicitTriangulation::preconditionVertexRankArray() {
-  this->vertexRankArray_.resize(this->vertexNumber_);
-  if(ttk::isRunningWithMPI()) {
-    ttk::produceRankArray(this->vertexRankArray_, this->vertGid_,
-                          this->vertexGhost_, this->vertexNumber_,
-                          this->boundingBox_.data(), this->neighborRanks_);
+  if(vertexRankArray_.size() == 0) {
+    this->vertexRankArray_.resize(this->vertexNumber_);
```

Comment:
I believe you want to write `this->vertexRankArray_.resize(this->vertexNumber_, 0);`
(to init it with 0, useful if not running with MPI)